containers/Ideas/ExplorePage.js

```diff
@@ -1,18 +1,30 @@
 import React, { useState, useEffect } from 'react'
 import { View, FlatList } from 'react-native'
 import API from '../../BaseApi'
+import AsyncStorage from '@react-native-async-storage/async-storage'
 import { ExploreListItem } from './ExploreListItem'
 import { styles } from './ExplorePage.styles'
 import { ButtonSignOut } from '../../components/ButtonSignOut'
 import { TextSourceSans } from '../../components/TextSourceSans'
+import {useAuthorization} from '../../containers/Auth/AuthProvider.js'
 
 export const ExplorePage = (props) => {
   const [projects, setProjects] = useState([])
+  const {signOut} = useAuthorization()
 
   const fetchProjects = () => {
-    API.getProjects()
-      .then(response => response && setProjects(response))
-      .catch(error => console.warn(error))
+    AsyncStorage.getItem('authToken')
+      .then((token) => API.getProjects(token))
+      .then((response) => {
+        if(response.statusCode === 200) {
```

Comment:
I know I did it like this but now that I look at it: should the fetch error code handling happen in BaseApi? and it either returns a results or an error? instead of doing the same error handling spread out over all the different containers?

Reply:
I actually tried to move the checking for the 401 to BaseApi, but I could not make it work to use the signOut there (because the useAuthentication is apparently a hook (why?)  and hook usage is only allowed in classes) :see_no_evil: So yes, I think it would be better, but not sure how to do it best.. do you want to give it a go @goapunk? Or we look at it together?

Reply:
ah, right. No, I don't think we can work around that as `BaseApi` is just plain js and not React